src/Plugin_TEST.cc

```diff
@@ -0,0 +1,248 @@
+  EXPECT_EQ("filename", plugin2.Filename());
+
+  EXPECT_EQ("pluginname", plugin.Name());
+  EXPECT_EQ("filename", plugin.Filename());
+}
+
+/////////////////////////////////////////////////
+TEST(DOMPlugin, CopyAssigmentOperator)
+{
+  sdf::Plugin plugin;
+  plugin.SetName("pluginname");
+  plugin.SetFilename("filename");
+
+  sdf::Plugin plugin2;
+  plugin2 = plugin;
+  EXPECT_EQ("pluginname", plugin2.Name());
+  EXPECT_EQ("filename", plugin2.Filename());
+
+  EXPECT_EQ("pluginname", plugin.Name());
+  EXPECT_EQ("filename", plugin.Filename());
+}
+
+/////////////////////////////////////////////////
+TEST(DOMPlugin, MoveAssignmentConstructor)
+{
+  sdf::Plugin plugin;
+  plugin.SetName("pluginname");
+  plugin.SetFilename("filename");
+
+  sdf::Plugin plugin2;
+  plugin2 = std::move(plugin);
+  EXPECT_EQ("pluginname", plugin2.Name());
+  EXPECT_EQ("filename", plugin2.Filename());
+}
+
+/////////////////////////////////////////////////
+TEST(DOMPlugin, CopyAssignmentAfterMove)
+{
+  sdf::Plugin plugin;
+  plugin.SetName("pluginname");
+  plugin.SetFilename("filename");
+
+  sdf::Plugin plugin2;
+  plugin2.SetName("pluginname2");
+  plugin2.SetFilename("filename2");
+
+  // This is similar to what std::swap does except it uses std::move for each
+  // assignment
+  sdf::Plugin tmp = std::move(plugin);
+  plugin = plugin2;
+  plugin2 = tmp;
+
+  EXPECT_EQ("pluginname", plugin2.Name());
+  EXPECT_EQ("filename", plugin2.Filename());
+
+  EXPECT_EQ("pluginname2", plugin.Name());
+  EXPECT_EQ("filename2", plugin.Filename());
+}
+
+/////////////////////////////////////////////////
+TEST(DOMPlugin, Load)
+{
+  sdf::Plugin plugin;
+  sdf::Errors errors;
+
+  // Null sdf
+  errors = plugin.Load(nullptr);
+  ASSERT_EQ(1u, errors.size());
+  EXPECT_EQ(sdf::ErrorCode::ELEMENT_MISSING, errors[0].Code());
+
+  // Bad element name
+  sdf::ElementPtr sdf(new sdf::Element());
+  sdf->SetName("bad");
+  errors = plugin.Load(sdf);
+  ASSERT_EQ(1u, errors.size());
+  EXPECT_EQ(sdf::ErrorCode::ELEMENT_INCORRECT_TYPE, errors[0].Code());
+  EXPECT_NE(nullptr, plugin.Element());
+
+  sdf->SetName("plugin");
+
+  // Missing name and filename attribute
+  errors = plugin.Load(sdf);
+  ASSERT_EQ(2u, errors.size());
+  EXPECT_EQ(sdf::ErrorCode::ATTRIBUTE_MISSING, errors[0].Code());
+  EXPECT_EQ(sdf::ErrorCode::ATTRIBUTE_MISSING, errors[1].Code());
+
+  sdf->AddAttribute("name", "string", "__default__", true);
+  sdf->GetAttribute("name")->Set<std::string>("my-plugin-name");
+
+  // Now just missing filename
+  errors = plugin.Load(sdf);
+  ASSERT_EQ(1u, errors.size());
+  EXPECT_EQ(sdf::ErrorCode::ATTRIBUTE_MISSING, errors[0].Code());
+
+  sdf->AddAttribute("filename", "string", "__default__", true);
+  sdf->GetAttribute("filename")->Set<std::string>("filename.so");
+
+  // No errors.
+  errors = plugin.Load(sdf);
+  ASSERT_TRUE(errors.empty());
+}
+
+/////////////////////////////////////////////////
+TEST(DOMPlugin, LoadWithChildren)
```

Comment:
check contents here ?

Reply:
That's tested through the `ToElement` function call in this test, which populates the `elem` object with the `contents` and does a string comparison of the result.